### PR TITLE
Add gold price data source

### PR DESCRIPTION
**Pull request by Benzbeeb**

### CODE REVIEW

**Review by sorawit — DISMISSED** [review]
Looks great

File: chain/datasources/binance_atom_price.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
```
Comment (taobun) [inline chain/datasources/binance_atom_price.sh]: Why we need this script? Using old binance script?

File: owasm/chaintests/gold_price/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+struct Output {
+    px: u64,
+}
+
+#[no_mangle]
+fn prepare_impl(input: Input) {
+    // Gold price data source
+    oei::request_external_data(5, 1, input.symbol.as_bytes());
+    // Binance data source
+    oei::request_external_data(6, 2, input.symbol.as_bytes());
```
Comment (taobun) [inline owasm/chaintests/gold_price/src/lib.rs]: I think we can use existing Binance script and send "ATOM".as_bytes() here instead

File: owasm/chaintests/gold_price/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use owasm::{execute_entry_point, ext, oei, prepare_entry_point};
+
+#[derive(BorshDeserialize)]
+struct Input {
+    symbol: String,
```
Comment (taobun) [inline owasm/chaintests/gold_price/src/lib.rs]: Don't need it in this case

**Review by taobun — CHANGES_REQUESTED** [review]

File: owasm/chaintests/gold_price/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+#[derive(BorshSerialize)]
+struct Output {
+    px: u64,
+}
+
+#[no_mangle]
+fn prepare_impl(input: Input) {
+    // Gold price data source
+    oei::request_external_data(5, 1, input.symbol.as_bytes());
```
Comment (taobun) [inline owasm/chaintests/gold_price/src/lib.rs]: Help me test if we send empty bytes, it will work correctly

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/app/genesis.go [hunk 5]
```diff
@@ -131,6 +136,11 @@ func GetDefaultDataSourcesAndOracleScripts(owner sdk.AccAddress) json.RawMessage
 			"Oracle script for getting an average crypto price from many sources encoding parameter by borsh.",
 			"./owasm/res/crypto_price_borsh.wasm",
 		},
+		{
+			"Gold price script",
+			"Oracle script for getting an average gold price from many sources.",
```
Comment (taobun) [inline chain/app/genesis.go]: Many sources LUL

Comment (taobun) [reply chain/app/genesis.go]: Oracle script for getting an average gold price in ATOM

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**